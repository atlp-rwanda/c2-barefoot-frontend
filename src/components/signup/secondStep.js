import React, { useState } from 'react';
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types';
import { Formik, Form, Field, useFormikContext } from 'formik';
import { TextField, Select, InputAdornment } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl'
import Upload from './imageUploadModel'
import * as yup from "yup";
import Box from '@material-ui/core/Box';
import { AccountCircle, Language, BusinessCenter, CloudUpload, Subject, EventBusy } from "@material-ui/icons";
import ListItemIcon from '@material-ui/core/ListItemIcon';
import axios from 'axios'
import Avatar from '@material-ui/core/Avatar';


const useStyles = makeStyles(theme => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    '@media(max-width: 320px)' : {
      display: 'grid'
    },
  },
  button: {
    margin: theme.spacing(1)
  },
  large: {
    width: theme.spacing(7),
    height: theme.spacing(7),
  },
  button: {
    margin: theme.spacing(1),
  },
  label: {
    display: 'inline-flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#d5d5d5',
    color: 'black',
    padding: '0.3rem',
    fontFamily: 'sans-serif',
    borderRadius: '0.3rem',
    cursor: 'pointer',
    margin: '1rem',
  },
  labelText: {
  }
}));

export  const FormPersonalDetails = ({
  formData,
  setFormData,
  nextStep,
  prevStep
}) => {
  const classes = useStyles();
  const [direction, setDirection] = useState('back');
  const [photo, setPhoto] = useState('');
  const [loading, setLoading] = useState(false);
  return (
    <>
      <Formik
        initialValues={formData}
        onSubmit={values => {
          setFormData(values);
          direction === 'back' ? prevStep() : nextStep();
        }}
      >
        {({ errors, touched, setFieldValue }) => (
          <Form className={classes.form}>
            <Box display='inline-flex' justifyContent='center' alignItems='center'>
              <Avatar 
              alt="user" 
              name='profile_picture' 
              margin='normal'
              src={formData ? formData.profile_picture : ''} 
              className={classes.large}/>
              {/* <Upload/> */}
              <input 
              id="file" 
              name="profile_picture" 
              type="file" 
              hidden='true'
              onChange={(e) => {
                const profile_picture = e.target.files[0]
                const formData = new FormData()
                formData.append('upload_preset', 'l9dhzfdi')
                formData.append('file', profile_picture)
                setLoading(true)
                
                axios.post(' https://api.cloudinary.com/v1_1/mjackson/image/upload', formData)
                .then(res => setFieldValue("profile_picture", res.data.secure_url))
                .then(
                  setLoading(false)
                )
                .catch(err => console.log(err))
              }}
              />
              <label for="file" className={classes.label}>
                <i class="fa fa-cloud-upload" aria-hidden="true"></i> 
                <div className={classes.labelText}>
                Upload Profile Image
                </div>

              </label>
            </Box>
            <Field 
            name='bio' 
            label='Biography' 
            as={TextField} 
            margin='normal'
              InputProps={{
                startAdornment: (
                  <InputAdornment position="start">
                    <Subject />
                  </InputAdornment>
                ),
              }}
            />
            <Field 
            name='occupation' 
            label='Occupation' 
            as={TextField} 
            margin='normal'
              InputProps={{
                startAdornment: (
                  <InputAdornment position="start">
                    <EventBusy />
                  </InputAdornment>
                ),
              }}
            />
            <Field 
            as={Select} 
            name="address" 
            margin='normal' 
            label='Address' 
            placeholder='Address'
            margin='normal'
              InputProps={{
                startAdornment: (
                  <ListItemIcon>
                    <AccountCircle />
                  </ListItemIcon>
                ),
              }}
            >
              <option selected value="kk 509 st">kk 509 st</option>
              <option value="kk 309 st">kk 309 st</option>
              <option value="kk 280 st">kk 280 st</option>
            </Field>
            <Field 
            as={Select} 
            name="language" 
            margin='normal' 
            label='Language' 
            placeholder='Language'
              InputProps={{
                startAdornment: (
                  <InputAdornment position="start">
                    <Language />
                  </InputAdornment>
                ),
              }}
            >
              <option selected value="Kinyarwanda">Kinyarwanda</option>
              <option value="English">English</option>
              <option value="Franch">Franch</option>
            </Field>
            <div>
              <Button
                type='submit'
                variant='contained'
                color='primary'
                className={classes.button}
                onClick={() => {
                  setDirection('back'); 
                  console.log(JSON.stringify(formData))
                }}
              >
                Back
              </Button>
              <Button
                type='submit'
                variant='contained'
                color='primary'
                className={classes.button}
                onClick={() => setDirection('forward')}
              >
                Next
              </Button>
            </div>
          </Form>
        )}
      </Formik>
    </>
  );
};

FormPersonalDetails.propTypes = {
  formData: PropTypes.object.isRequired,
  setFormData: PropTypes.func.isRequired,
  nextStep: PropTypes.func.isRequired,
  prevStep: PropTypes.func.isRequired
};
