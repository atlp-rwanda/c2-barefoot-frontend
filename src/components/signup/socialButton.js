import React, {Component} from 'react'
import FacebookLoginWithButton from 'react-facebook-login';
import GoogleLoginButton from 'react-google-login'


const responseGoogle = (response) => {
    console.log(response);
  }
class SocialButtons extends Component{
    render(){
        return(
            
            (this.props.stepIndex == 0 ? (
                <>
                    <p>Or</p>
                    
                    <FacebookLoginWithButton
                        appId="1206715649505081"
                        textButton='Signup with facebook'
                        size= 'small'
                        fields="name,email,picture"
                        icon="fa-facebook"/>
                    <p>Or</p>
                    <GoogleLoginButton
                        clientId="658977310896-knrl3gka66fldh83dao2rhgbblmd4un9.apps.googleusercontent.com"
                        // render={renderProps => (
                        //   <button onClick={renderProps.onClick} disabled={renderProps.disabled}>This is my custom Google button</button>
                        // )}
                        buttonText="SIGNUP WITH GOOGLE"
                        onSuccess={responseGoogle}
                        onFailure={responseGoogle}
                        cookiePolicy={'single_host_origin'}
                    />
                </>
            ): '')
        )
    }
}

export default SocialButtons