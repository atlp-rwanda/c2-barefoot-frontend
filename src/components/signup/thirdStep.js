import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';
import { List, ListItem, ListItemText } from '@material-ui/core/';
import { AccountCircle, Email, BusinessCenter, Language, Subject, EventBusy } from "@material-ui/icons";
import ListItemIcon from '@material-ui/core/ListItemIcon';
import Avatar from '@material-ui/core/Avatar';
import axios from 'axios';
import Snackbar from '@material-ui/core/Snackbar';
import MuiAlert from '@material-ui/lab/Alert';


const useStyles = makeStyles(theme => ({
  container: {
    textAlign: 'left'
  },
  textCenter: {
    textAlign: 'center'
  },
  button: {
    margin: theme.spacing(1)
  },
  large: {
    width: theme.spacing(10),
    height: theme.spacing(10),
  }
}));


export const Confirm = ({ formData, prevStep, nextStep }) => {
  const classes = useStyles();
  const { first_name, last_name, email, username, occupation, bio, address, language, profile_picture} = formData;
  const [errorOpen, setErrorOpen] = React.useState(false);
  const [successOpen, setSuccessOpen] = React.useState(false);
  const [success, setSuccess] = React.useState('');
  const [error, setError] = React.useState('');

  const signupRequest = () => {
    const{confirmPassword, ...user} = {...formData}
    axios.post(`https://barefoot-nomad-app-v1.herokuapp.com/api/v1/user/signup`, user)
    .then(res => {
      setSuccessOpen(true)
      nextStep()
    })
    .catch(err => { 
      if (err.response){
        setError(err.response.data.error)
        setErrorOpen(true);
      }else if(err.request){
        setError(err.request.data.error)
        setErrorOpen(true)
      }else if(err.message){
        setError(err.message.data.error)
        setErrorOpen(true)      
      }
    })
  }
  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }
    setErrorOpen(false);
  };
  function Alert(props) {
    return <MuiAlert elevation={6} variant="filled" {...props} />;
  }
  return (
    <>
      <Snackbar open={errorOpen} autoHideDuration={6000} onClose={handleClose}>
        <Alert onClose={handleClose} severity="error">
          Error: {error ? JSON.stringify(error).replace(/[\\'"]+/g, '') : 'Error Not set'}
        </Alert>
      </Snackbar>
      <Snackbar open={successOpen} autoHideDuration={6000} onClose={handleClose}>
        <Alert onClose={handleClose} severity="success">
          Success: {success ? JSON.stringify(success) : 'Message Not set'}
        </Alert>
      </Snackbar>
      <div>
        <List>
        <ListItem>
            <Avatar alt="user" name='file' src={profile_picture ? profile_picture : ''} className={classes.large}/>
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <AccountCircle />
            </ListItemIcon>
            <ListItemText
              primary='Names'
              secondary={first_name + ' ' + last_name}
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <AccountCircle />
            </ListItemIcon>
            <ListItemText
              primary='Username'
              secondary={username}
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <Email/>
            </ListItemIcon>
            <ListItemText
              primary='Email'
              secondary={email}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <AccountCircle />
            </ListItemIcon>
            <ListItemText
              primary='Occupation'
              secondary={occupation}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <BusinessCenter />
            </ListItemIcon>
            <ListItemText
              primary='Address'
              secondary={address}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <Language />
            </ListItemIcon>
            <ListItemText
              primary='Language'
              secondary={language}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <Subject />
            </ListItemIcon>
            <ListItemText
              primary='Biography'
              secondary={bio}
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <EventBusy />
            </ListItemIcon>
            <ListItemText
              primary='Occupation'
              secondary={occupation}
            />
          </ListItem>
        </List>
        <div className={classes.textCenter}>
          <Button
            color='secondary'
            variant='contained'
            id="backBtn"
            className={classes.button}
            onClick={() => prevStep()}
          >
            Back
          </Button>

          <Button
            color='primary'
            variant='contained'
            // id="submitBtn_"
            btn='submitBtn'
            className={classes.button}
            onClick={() => signupRequest()}
          >
            Submit
          </Button>
        </div>
      </div>
    </>
  );
};

Confirm.propTypes = {
  formData: PropTypes.object.isRequired,
  prevStep: PropTypes.func.isRequired,
  nextStep: PropTypes.func.isRequired
};
