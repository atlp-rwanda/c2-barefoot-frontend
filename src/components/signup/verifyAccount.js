import { Box, Button, makeStyles } from '@material-ui/core'
import Devider from '@material-ui/core/Divider'
import { BorderBottom, CheckCircle } from '@material-ui/icons'
import Axios from 'axios'
import React, {Component, useState} from 'react'
import { Link, useParams } from 'react-router-dom'

const useStyle = makeStyles(theme => ({
    container: {
        display: 'grid',
        justifyContent: 'center',
        flexDirection: 'column',
        alignItems: 'center',
        height: '400px',
        padding: '10px'
    },
    boxContainer: {
        textAlign: 'center'
    },
    box: {
        display: 'flex',
        justifyContent: 'center',
        // flexDirection: 'row',
        alignItems: 'center',
        padding: '10px',
        color: '#90a4ae'
    },
    hr: {
        height: '2px',
        margin: '5px -10px',
        '@media(max-width: 740px)': {
            margin: '5px'
        }
    },
    tokenMissing: {
        fontSize: '5vh',
        padding: '20px',
        borderBottom: '2px black solid',
    }
}))

const verifyAccount = () => {
    const classes = useStyle()
    const { token} = useParams();
    const [verified, setVerified] = useState(false)
    const [error, setError] = useState('');
    console.log("Token: " + token)
    
    Axios.patch(`https://barefoot-nomad-app-v1.herokuapp.com/api/v1/user/verification?token=` + token)
    .then(res => {
        console.log(JSON.stringify(verified))
      setVerified(true)
    })
    .catch(err => { 
      if (err.response){
        setError(err.response.data.error)
      }else if(err.request){
        setError(err.request.data.error)
      }else if(err.message){
        setError(err.message.data.error)
      }
    })
    return (
        
            <div className={classes.container}>
                <Box className={classes.boxContainer}>
                    {token && verified ? (
                    <>
                        <Box className={classes.box}>
                            <CheckCircle style={{color: '#1b5e20'}}/>Your e-mail has been verified
                            {"Verified : " + JSON.stringify(verified)}
                        </Box>
                        <Devider flexItem className={classes.hr}/>
                        <Box className={classes.box}>
                            Thank you for verifying your account. You can now continue to use BarefootNomad
                        </Box>
                        
                    </>) : (
                        <Box className={classes.tokenMissing}>
                            :( Token Missing
                        </Box>
                    )}
                    <Button variant="contained" color="primary" href='/'> Go Home</Button>
                </Box>
                
            </div>
        
        
    )
}

export default verifyAccount