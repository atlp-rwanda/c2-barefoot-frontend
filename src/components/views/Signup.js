import React from 'react'
import SignUp from '../signup'

export default function Signup(){
    return(
        <SignUp/>
    )
}