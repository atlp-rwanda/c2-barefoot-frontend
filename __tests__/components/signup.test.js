import { shallow, mount } from "enzyme";
import React from 'react';
import toJson from "enzyme-to-json";
import { FirstStep } from "../../src/components/signup/firstStep";
import { SecondStep } from "../../src/components/signup/secondStep";
import Signup from '../../src/components/signup'
import { Confirm } from "../../src/components/signup/thirdStep";
import verifyAccount from "../../src/components/signup/verifyAccount";
import SocialButtons from '../../src/components/signup/socialButton'
import SideDiv from "../../src/components/signup/sideDiv";

describe('Signup', () => {
    let formData = new FormData()

    const props = {
        formData:formData,
        setFormData: function(){},
        nextStep: function(){},
        prevStep: function(){},
        profile_picture: 'image.jpg'
    }
    describe('Signup Rendering', ()=>{
        
        it('renders correctly first step', () => {
            const wrapper = shallow(<FirstStep {...props}/>)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('renders correctly Signup index', () => {
            const wrapper = shallow(<Signup {...props}/>)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('renders correctly Third step', () => {
            const wrapper = shallow(<Confirm {...props}/>)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('renders side div correctly', () => {
            const wrapper = shallow(<SideDiv />)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('renders Social buttons correctly', () => {
            const wrapper = shallow(<SocialButtons />)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        
    })
    describe('Functionalities', () => {
        it('should call onChange prop with input value', () => {
            const onSearchMock = jest.fn();
            const component = mount(<SecondStep onChange={onSearchMock} {...props} value="custom value" />);
            component.find('#file').simulate('change');
            // expect(onSearchMock).toBeCalledWith('image.jpg');
        });
        it('should call onClick to submit form', () => {
            const signupRequest = jest.fn();
            const handleClose = jest.fn();
            const component = mount(<Confirm onClick={signupRequest} handleClose={handleClose} {...props} value="custom value" />);
            component.find('[btn="submitBtn"]').first().simulate('click');
            // expect(onSearchMock).toBeCalledWith('image.jpg');
        });
    })

})
