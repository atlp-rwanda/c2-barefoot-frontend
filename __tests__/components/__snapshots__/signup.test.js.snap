// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Signup Rendering renders Social buttons correctly 1`] = `""`;

exports[`Signup Signup Rendering renders correctly Signup index 1`] = `
<div
  className="makeStyles-container-11"
>
  <SideDiv
    loading={true}
  />
  <div
    className="makeStyles-root-8"
  >
    <div
      className="makeStyles-rightSide-9"
    >
      <Styled(MuiBox)
        className="makeStyles-box-15"
      >
        <WithStyles(ForwardRef(Skeleton))
          className="makeStyles-textBox-14"
          variant="rect"
        />
      </Styled(MuiBox)>
      <Styled(MuiBox)
        className="makeStyles-stepperContainer-16"
      >
        <Styled(MuiBox)
          className="makeStyles-stepperBox-17"
        >
          <WithStyles(ForwardRef(Skeleton))
            variant="circle"
          >
            <WithStyles(ForwardRef(Avatar)) />
          </WithStyles(ForwardRef(Skeleton))>
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
        </Styled(MuiBox)>
        <Styled(MuiBox)
          className="makeStyles-stepperBox-17"
        >
          <WithStyles(ForwardRef(Skeleton))
            variant="circle"
          >
            <WithStyles(ForwardRef(Avatar)) />
          </WithStyles(ForwardRef(Skeleton))>
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
        </Styled(MuiBox)>
        <Styled(MuiBox)
          className="makeStyles-stepperBox-17"
        >
          <WithStyles(ForwardRef(Skeleton))
            variant="circle"
          >
            <WithStyles(ForwardRef(Avatar)) />
          </WithStyles(ForwardRef(Skeleton))>
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
        </Styled(MuiBox)>
      </Styled(MuiBox)>
      <div>
        <div>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-instructions-13"
          >
            <FirstStep
              formData={
                Object {
                  "address": "",
                  "bio": "",
                  "email": "",
                  "first_name": "",
                  "language": "",
                  "last_name": "",
                  "occupation": "",
                  "password": "",
                  "profile_picture": "",
                  "username": "",
                }
              }
              loading={true}
              nextStep={[Function]}
              setFormData={[Function]}
            />
          </WithStyles(ForwardRef(Typography))>
        </div>
      </div>
    </div>
    <SocialButtons
      loading={true}
      stepIndex={0}
    />
  </div>
</div>
`;

exports[`Signup Signup Rendering renders correctly Third step 1`] = `
<Fragment>
  <Loader
    open={false}
  />
  <WithStyles(ForwardRef(Snackbar))
    autoHideDuration={6000}
    onClose={[Function]}
    open={false}
  >
    <Alert
      onClose={[Function]}
      severity="error"
    >
      Error: 
      Error Not set
    </Alert>
  </WithStyles(ForwardRef(Snackbar))>
  <WithStyles(ForwardRef(Snackbar))
    autoHideDuration={6000}
    onClose={[Function]}
    open={false}
  >
    <Alert
      onClose={[Function]}
      severity="success"
    >
      Success: 
      Message Not set
    </Alert>
  </WithStyles(ForwardRef(Snackbar))>
  <div>
    <WithStyles(ForwardRef(List))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(Avatar))
          alt="user"
          className="makeStyles-large-22"
          name="file"
          src=""
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo([object Object]) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Names"
          secondary="undefined undefined"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo([object Object]) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Username"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo([object Object]) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Email"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo([object Object]) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Occupation"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo([object Object]) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Address"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo([object Object]) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Language"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo([object Object]) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Biography"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo([object Object]) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Occupation"
        />
      </WithStyles(ForwardRef(ListItem))>
    </WithStyles(ForwardRef(List))>
    <div
      className="makeStyles-textCenter-20"
    >
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-21"
        color="secondary"
        id="backBtn"
        onClick={[Function]}
        variant="contained"
      >
        Back
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        btn="submitBtn"
        className="makeStyles-button-21"
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        Submit
      </WithStyles(ForwardRef(Button))>
    </div>
  </div>
</Fragment>
`;

exports[`Signup Signup Rendering renders correctly first step 1`] = `
<Fragment>
  <Formik
    initialValues={FormData {}}
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_excludedEdges": Array [],
        "_mutate": undefined,
        "_nodes": Array [
          "password",
          "confirmPassword",
          "email",
          "username",
          "last_name",
          "first_name",
        ],
        "_sortErrors": [Function],
        "_typeError": [Function],
        "_whitelist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "conditions": Array [],
        "deps": Array [],
        "exclusiveTests": Object {},
        "fields": Object {
          "confirmPassword": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [
              Condition {
                "fn": [Function],
                "refs": Array [
                  Reference {
                    "getter": [Function],
                    "isContext": false,
                    "isSibling": true,
                    "isValue": false,
                    "key": "password",
                    "map": undefined,
                    "path": "password",
                  },
                ],
              },
            ],
            "deps": Array [
              "password",
            ],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "email": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "email": false,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "first_name": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "max": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "last_name": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "max": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "password": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "min": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "username": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "max": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "spec": Object {
          "abortEarly": true,
          "label": undefined,
          "meta": undefined,
          "nullable": false,
          "presence": "optional",
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  >
    <Component />
  </Formik>
</Fragment>
`;

exports[`Signup Signup Rendering renders side div correctly 1`] = `
<div
  className="makeStyles-sideNav-23"
>
  <div
    className="makeStyles-sideText-27"
  >
    <p>
      Please create account to get 
    </p>
    <p>
      started with Barefoot Nomard.
    </p>
  </div>
  <div
    className="makeStyles-img-25"
  >
    <img
      alt="Barefoot Nomards"
      className="makeStyles-locationImg-24"
      src="../public/images/location.png"
    />
  </div>
</div>
`;
